### PR TITLE
Drop old testsuite

### PR DESCRIPTION
**Pull request by jreidinger**
trello: https://trello.com/c/lmE34oq8/1021-drop-old-testsuite

what this PR do:

- convert/drop old testsuite
- switch from autotools to rake ( non trivial as tests does not work in build root )
- enable coveralls

### CODE REVIEW

File: test/sysconfig_test.rb [hunk 1]
```diff
@@ -9,6 +9,23 @@
...
   end
+  let(:configfiles) { ["#{DATA_PATH}/sysconfig/*"] }
+
+  before do
+    allow(Yast::SCR).to receive(:Execute).with(path(".target.bash_output"), anything)
+      .and_return("exit" => 0, "stdout" => "", "stderr" => "")
+    # run git version of parse_configs.pl
+    allow(Yast::SCR).to receive(:Execute).with(path(".target.bash"), /parse_configs.pl/) do |path, command|
+      params = command.split
+      params[0] = SRC_PATH + "/bin/parse_configs.pl"
```
Comment (lslezak) [inline test/sysconfig_test.rb]: NP: `File.join` would be a  bit better...
```suggestion
      params[0] = File.join(SRC_PATH, "bin/parse_configs.pl")
```

File: test/sysconfig_test.rb [hunk 2]
```diff
@@ -9,6 +9,23 @@
...
+      `#{params.join(" ")}`
+    end
+
+    allow(Yast::SCR).to receive(:Read).and_call_original
+    # read properly sysedit agent from git
+    allow(Yast::SCR).to receive(:Read).with(path(".target.string"), /sysedit.agent/)
+      .and_return(::File.read(SRC_PATH+"/data/sysedit.agent"))
```
Comment (lslezak) [inline test/sysconfig_test.rb]: Missing spacesaround `+`, Rubocop would complain... :wink: 

But again, `File.join` would be a bit better:
```suggestion
      .and_return(::File.read(File.join(SRC_PATH, "/data/sysedit.agent")))
```

File: test/sysconfig_test.rb [hunk 3]
```diff
@@ -213,4 +230,182 @@
...
+      expected = "test\n" \
+        "   \t\t\n" \
+        " The END\n"
+
+      expect(subject.get_only_comment(input)).to eq expected
+    end
+  end
+
+  describe ".parse_metadata" do
+    it "return empty hash if nil passed" do
```
Comment (lslezak) [inline test/sysconfig_test.rb]:
```suggestion
    it "returns empty hash if nil passed" do
```

File: test/sysconfig_test.rb [hunk 4]
```diff
@@ -213,4 +230,182 @@
...
+        expect(sysconfig.modified("#{var}$#{entry_path}")).to eq true
+      end
+
+      it "sets the Modified flag" do
+        expect(sysconfig.Modified).to eq true
+      end
+    end
+
+    context "with numeric random values" do
+      let(:var) { rand(8888) }
```
Comment (lslezak) [inline test/sysconfig_test.rb]: Do we really need to test random values? I mean I do not like any randomness in tests, randomly failing tests is the worst thing ever. If we need to test "random" values we can write some random static number there.

File: test/sysconfig_test.rb [hunk 5]
```diff
@@ -213,4 +230,182 @@
...
+      let(:value) { "NoWayThisIsCorrectValue" }
+
+      it "returns an identical entry" do
+        sysconfig.Import([autoyast_entry(var, value, entry_path)])
+        expect(sysconfig.Export).to eq [autoyast_entry(var, value, entry_path)]
+      end
+    end
+
+    context "with numeric random imported values" do
+      let(:var) { rand(8888) }
```
Comment (lslezak) [inline test/sysconfig_test.rb]: Here as well?

File: test/sysconfig_test.rb (hunk 4 above)
Comment (jreidinger) [reply test/sysconfig_test.rb]: this is not my test. I just copy what was in sysconfig_autoyast_test.rb to follow ruby convention.

**Pull request merged**